src/main/java/shon/Parser.java

```diff
@@ -0,0 +1,225 @@
+package shon;
+
+import shon.command.AddDeadlineCommand;
+import shon.command.AddEventCommand;
+import shon.command.AddTodoCommand;
+import shon.command.Command;
+import shon.command.DeleteTaskCommand;
+import shon.command.ExitCommand;
+import shon.command.FindCommand;
+import shon.command.ListCommand;
+import shon.command.MarkCommand;
+import shon.command.UnmarkCommand;
+
+import shon.exception.CommandException;
+import shon.exception.ParameterException;
+
+/**
+ * Represents a Parser that makes sense of the user's inputs.
+ */
+public class Parser {
+    /** The allowed set of actions that the user can do */
+    private enum Action {
+        LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, BYE
+    }
+
+    /**
+     * Parses the user's input into a command with the appropriate parameters.
+     *
+     * @param input User's input.
+     * @return Command to be executed.
+     * @throws CommandException If given user input is empty, or a command not allowed in <code>Action</code>.
+     * @throws ParameterException If given parameters for that command is invalid.
+     */
+    public static Command parse(String input) throws CommandException, ParameterException {
+        Action action = Parser.getAction(input);
+        switch (action) {
+        case LIST:
```

Comment:
consider adding break for each case as shown in the coding standards

Reply:
Hm I was considering if it was needed since all my cases have a return statement. Putting a break after the return statement seems quite pointless and the IDE flags the break statement as an unreachable statement and recommends it to be removed.